corehq/apps/reports/util.py

```diff
@@ -602,3 +626,83 @@ def _add_new_user_to_HQ_group(session, local_user):
         _sync_tableau_users_with_hq(domain)
         # Sync the TableauUser model with Tableau users on the remote Tableau instance
         _sync_tableau_users_with_remote(domain)
+
+
+# Attaches to the parse_web_users method
+def add_on_tableau_details(domain, web_user_dicts):
+
+    session = TableauAPISession.create_session_for_domain(domain)
+
+    def _get_roles_by_user(domain, usernames):
+        return {
+            user['username']: user['role'] for user in TableauUser.objects.filter(
+                server=TableauServer.objects.get(domain=domain)
+            ).filter(username__in=usernames).values('username', 'role')
+        }
+
+    def _get_groups_by_user(session, domain, usernames):
+        try:
+            all_groups = get_all_tableau_groups(domain, session=session)
+            users_by_group = {}
+            for group in all_groups:
+                users_by_group[group] = session.get_users_in_group(group.id)
+        except TableauAPIError:
+            return -1
+
+        def _username_in_user_dicts(username, user_dicts):
+            # Is there at least one user_dict with the given username
+            return bool(sum([user_dict.get('name') == tableau_username(username) for user_dict in user_dicts]))
```

Comment:
a simpler way to do this is to use `any` which is also more efficient since it will stop evaluating once it finds a non false item:
```
any(user_dict.get('name') == tableau_username(username) for user_dict in user_dicts)
```